packages/common/src/hooks/purchaseContent/usePurchaseContentErrorMessage.ts

```diff
@@ -0,0 +1,37 @@
+import { BuyUSDCErrorCode } from 'store/index'
+import {
+  PurchaseContentErrorCode,
+  PurchaseErrorCode
+} from 'store/purchase-content'
+import { formatPrice } from 'utils/formatUtil'
+
+import { RemoteVarHook } from '../useRemoteVar'
+
+import { useUSDCPurchaseConfig } from './useUSDCPurchaseConfig'
+
+const messages = {
+  generic: 'Your purchase was unsuccessful.',
+  minimumPurchase: (minAmount: number) =>
+    `Total purchase amount must be at least $${formatPrice(minAmount)}.`,
+  maximumPurchase: (maxAmount: number) =>
+    `Total purchase amount may not exceed $${formatPrice(maxAmount)}.`
+}
+
+export const usePurchaseContentErrorMessage = (
+  errorCode: PurchaseContentErrorCode,
+  useRemoteVar: RemoteVarHook
+) => {
+  const { minUSDCPurchaseAmountCents, maxUSDCPurchaseAmountCents } =
+    useUSDCPurchaseConfig(useRemoteVar)
+
+  switch (errorCode) {
+    case BuyUSDCErrorCode.MinAmountNotMet:
+      return messages.minimumPurchase(minUSDCPurchaseAmountCents)
+    case BuyUSDCErrorCode.MaxAmountExceeded:
+      return messages.maximumPurchase(maxUSDCPurchaseAmountCents)
+    case BuyUSDCErrorCode.OnrampError:
+    case PurchaseErrorCode.Unknown:
+    default:
```

Comment:
In TS you don't need the default cause, and I like not including it because it forces us to make sure we're intentional about what message we show to users if we add a new error type

Reply:
Oh interesting, will update. I agree that it's nice to have the compiler complain when you don't implement a newly-added case.